xapian-core/api/queryinternal.cc

```diff
@@ -865,98 +867,156 @@ Query::Internal::unserialise(const char ** p, const char * end,
+			}
+			return new Xapian::Internal::QueryWildcard(
+					pattern,
+					max_expansion,
+					flags,
+					combiner);
 		    }
-		    return new Xapian::Internal::QueryWildcard(pattern,
-							       max_expansion,
-							       flags,
-							       combiner);
-		}
-		case 0x0c: { // PostingSource
-		    string name;
-		    if (!unpack_string(p, end, name)) {
-			throw SerialisationError("not enough data");
+		    case 0x0c: { // PostingSource
+			string name;
+			if (!unpack_string(p, end, name)) {
+			    throw SerialisationError("not enough data");
+			}
+
+			const PostingSource* reg_source =
+					reg.get_posting_source(name);
+			if (!reg_source) {
+			    string m = "PostingSource ";
+			    m += name;
+			    m += " not registered";
+			    throw SerialisationError(m);
+			}
+
+			string serialised_source;
+			if (!unpack_string(p, end,
+					   serialised_source)) {
+				throw SerialisationError(
+						"not enough data");
+			}
+			PostingSource* source =
+			    reg_source->unserialise_with_registry(
+						serialised_source,
+						reg);
+			return new Xapian::Internal::QueryPostingSource
+						(source->release());
 		    }
-
-		    const PostingSource * reg_source = reg.get_posting_source(name);
-		    if (!reg_source) {
-			string m = "PostingSource ";
-			m += name;
-			m += " not registered";
-			throw SerialisationError(m);
+		    case 0x0d: {
+			using Xapian::Internal::QueryScaleWeight;
+			double scale_factor = unserialise_double(p,
+								 end);
+			return new QueryScaleWeight(scale_factor,
+			    Query(unserialise(p, end, reg)));
 		    }
-
-		    string serialised_source;
-		    if (!unpack_string(p, end, serialised_source)) {
-			throw SerialisationError("not enough data");
-		    }
-		    PostingSource* source =
-			reg_source->unserialise_with_registry(serialised_source,
-							      reg);
-		    return new Xapian::Internal::QueryPostingSource(source->release());
-		}
-		case 0x0d: {
-		    using Xapian::Internal::QueryScaleWeight;
-		    double scale_factor = unserialise_double(p, end);
-		    return new QueryScaleWeight(scale_factor,
-						Query(unserialise(p, end, reg)));
-		}
-		case 0x0e: {
-		    Xapian::termcount wqf;
-		    Xapian::termpos pos;
-		    if (!unpack_uint(p, end, &wqf) ||
-			!unpack_uint(p, end, &pos)) {
-			throw SerialisationError("not enough data");
+		    case 0x0e: {
+			Xapian::termcount wqf;
+			Xapian::termpos pos;
+			if (!unpack_uint(p, end, &wqf) ||
+			    !unpack_uint(p, end, &pos)) {
+				throw SerialisationError(
+						"not enough data");
+			}
+			return new Xapian::Internal::QueryTerm(
+						string(), wqf, pos);
 		    }
-		    return new Xapian::Internal::QueryTerm(string(), wqf, pos);
-		}
-		case 0x0f:
-		    return new Xapian::Internal::QueryTerm();
-		default: // Others currently unused.
-		    break;
+		    case 0x0f:
+			return new Xapian::Internal::QueryTerm();
+		    default: // Others currently unused.
+			break;
+			}
+	    }
 	    }
 	    break;
 	}
     }
+
```

Comment:
Please avoid introducing whitespace changes unless there's a good reason (I can't see one here). Even then, they would ideally be in a separate commit, which I don't think is true here.